### ISSUE TITLE
Cannot list arches supported through emulation

### ISSUE THREAD

**Issue by aleixpol** [flatpak/flatpak#736]
I set up my system to be able to build/execute arm binaries using `--arch=arm`, although `flatpak --supported-arches` still only lists i386 and x86_64.

**Comment by alexlarsson** [comment]
Yeah, that really tricky to figure out though. --supported arches is basically based on `uname -m`, i.e. the actual hardware architecture.

Do you really need it to list emulated architecures though? Seems like claiming those are suppored is partially lying.

**Comment by aleixpol** [comment]
I understand. I created the bug report because I was implementing IDE support, where I want to let the developers to interact with any-ish architecture.

If you think it's too much of a hack, I can list architectures based on which are available for a said Sdk too.

**Comment by aleixpol** [comment]
If listing is the solution, then there's also a problem, as `flatpak info` will only list local architectures:
```
apol@oliver:~$ flatpak info org.kde.Sdk//master -r
runtime/org.kde.Sdk/x86_64/master
apol@oliver:~$ flatpak info org.kde.Sdk//master -r --arch arm
runtime/org.kde.Sdk/arm/master

```

**Comment by alexlarsson** [comment]
I think we could take some kind of commandline argument which makes it also consider emulated architectures supported (because i don't really think this is right by default for regular users).

However, that doesn't really help me with the basic issue. In the flatpak code, how would it know if a emulation of a particular architecture works?

Maybe the easiest approach is just supporting some environment variable like FLATPAK_SUPPORTED_ARCHES, and then punt on the detection?

**Comment by aleixpol** [comment]
There must be a reason why a _regular user_ ended up installing arm binaries.

**Comment by alexlarsson** [comment]
True, in the case of locally installed files we can just not filter by arch. However, I was thinking of all that the supported arches is used for, which involves things like listing the contents of a remote.

But yeah, honestly I'm not sure there is ever a reason to filter what is shown of the locally installed apps. 
We could be using a shared homedir or /var between machines of multiple arches, but that seems very niche.

**Comment by alexlarsson** [comment]
Its unclear to me exactly what the issue is with listing.
For example:
```
$ flatpak --user list | grep org.freedesktop.Platform.*1\.6
org.freedesktop.Platform/arm/1.6                       runtime        
org.freedesktop.Platform/i386/1.6                      runtime        
org.freedesktop.Platform/x86_64/1.6                    runtime      
```
This seem to list the arm version.
Also, i can get info about it:
```
$ flatpak info org.freedesktop.Platform/arm/1.6
Ref: runtime/org.freedesktop.Platform/arm/1.6
ID: org.freedesktop.Platform
Arch: arm
Branch: 1.6
Origin: gnome
Commit: 009e63050d0387d5cc34e0bf9b2554396d9b71dc7290a6eff9ba7382358b73bb
Location: /home/alex/.local/share/flatpak/runtime/org.freedesktop.Platform/arm/1.6/009e63050d0387d5cc34e0bf9b2554396d9b71dc7290a6eff9ba7382358b73bb
Installed size: 450,8 MB
```

If i specify no arch then i get the current arch as the default:
```
flatpak info org.freedesktop.Platform/arm/1.6
Ref: runtime/org.freedesktop.Platform/arm/1.6
ID: org.freedesktop.Platform
Arch: arm
Branch: 1.6
Origin: gnome
Commit: 009e63050d0387d5cc34e0bf9b2554396d9b71dc7290a6eff9ba7382358b73bb
Location: /home/alex/.local/share/flatpak/runtime/org.freedesktop.Platform/arm/1.6/009e63050d0387d5cc34e0bf9b2554396d9b71dc7290a6eff9ba7382358b73bb
Installed size: 450,8 MB
[jhbuild ~] flatpak info org.freedesktop.Platform//1.6
Ref: runtime/org.freedesktop.Platform/x86_64/1.6
ID: org.freedesktop.Platform
Arch: x86_64
Branch: 1.6
Origin: gnome
Commit: 5c357b50c9aaaf26f2fe1be6f301e2236aef70c88493048805d5b63d43a8b3ee
Location: /home/alex/.local/share/flatpak/runtime/org.freedesktop.Platform/x86_64/1.6/5c357b50c9aaaf26f2fe1be6f301e2236aef70c88493048805d5b63d43a8b3ee
Installed size: 553,9 MB
```
But i think this makes sense.

**Comment by aleixpol** [comment]
My only criticism was that we can't list them with info. if that's what we're aiming for, then feel free to close the issue and I'll port it to `flatpak list` + filtering which will work as well.

Here on the second execution it could list both.
```
$ flatpak info -r org.kde.Sdk/arm/master
runtime/org.kde.Sdk/arm/master
$ flatpak info -r org.kde.Sdk//master
runtime/org.kde.Sdk/x86_64/master
```

**Comment by orowith2os** [comment]
Sorry to necro this, but recently came across this, and was wondering. Is it not possible to check binfmt_misc, or the appropriate APIs, for the supported binaries we can run? Are there even APIs for that? I wouldn't say that claiming to support emulated arches is "lying" either, you still support them - just not natively.

**Comment by C0ntr0lledCha0s** [comment]
Just to add to this, currently trying to run some amd64 apps on arm64 arch using Box64 as an emulator but not able to install through flatpak as it's not a supported arch.
